lib/dor/text_extraction/ocr.rb

```diff
@@ -94,20 +94,22 @@ def cleanup_input_folder
       def cleanup_output_folder
         return unless Dir.exist?(abbyy_output_path)
 
-        # the output folder *should* be empty by now, but sometimes it takes a bit longer for it to actually be
-        # empty or appear to be empty (file system sync issues)...since the rm -fr call does not actually seem
-        # to remove non-empty folders, wait a bit until the file system thinks the folder is empty,
-        # but eventually giving up so we don't wait forever
+        files = Dir.glob("#{abbyy_output_path}/*")
+        files.each { |file| FileUtils.rm_r(file) }
+        logger.info "Removing ABBYY output directory: #{abbyy_output_path}.  Empty: #{Dir.empty?(abbyy_output_path)}. Num files/folders: #{files.count}: #{files.join(', ')}"
+        # the output folder *should* be empty/deleted by now, but sometimes it takes a bit longer for it to actually be
+        # empty/deleted or appear to be empty/deleted (file system sync issues)
+        # wait a bit until the file system thinks the folder is empty, but eventually giving up so we don't wait forever
         tries = 0
-        max_tries = 5
+        max_tries = 7
         loop do
+          delete_folder(abbyy_output_path)
           tries += 1
-          sleep(3**tries)
-          break if Dir.empty?(abbyy_output_path) || tries > max_tries
+          break if !Dir.exist?(abbyy_output_path) || Dir.empty?(abbyy_output_path) || tries > max_tries
+
+          sleep(2**tries)
         end
 
-        files = Dir.glob("#{abbyy_output_path}/*")
-        logger.info "Removing ABBYY output directory: #{abbyy_output_path}.  Empty: #{Dir.empty?(abbyy_output_path)}. Num files/folders: #{files.count}: #{files.join(', ')}"
         delete_folder(abbyy_output_path)
```

Comment:
i have futzed with this for hours now, and this actually seems to clear the output folder

i remain mystified as to why it is all needed, but suffice to say that simply trying to force delete the folder, which should delete it and all of it's contents, does not work... nor does delete the files first, and then trying to delete the (what should be) empty folder

I presume it is file system sync issues and it's a real pain